### ISSUE TITLE
comma-dangle bug

### ISSUE THREAD

**Issue by adispring** [eslint/eslint#7851]
**Tell us about your environment**

* **ESLint Version: v3.12.2**
* **Node Version: v6.9.1**
* **npm Version: 3.10.9**

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

```
{
  "extends": "airbnb",
  "rules": {
    "strict": 0,
    "no-underscore-dangle": ["error", { "allow": ["__"] }],
    "new-cap": ["error", { "capIsNew": false }],
    "no-confusing-arrow": ["error", { "allowParens": true }]
  },
  "parserOptions": {
      "ecmaVersion": 6,
      "ecmaFeatures": {
          "experimentalObjectRestSpread": true
      }
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
        const headers = R.compose(
          R.assoc('X-Requested-With', 'XMLHttpRequest'),
          R.merge({ 'Content-Type': 'application/json;charset=UTF-8' }),
          R.defaultTo({})
        )(header);
```

**What did you expect to happen?**
in the line of `R.defaultTo({})`, should not output "comma-dangle" error

**What actually happened? Please include the actual, raw output from ESLint.**
Missing trailing comma. (comma-dangle)

**Comment by vitorbal** [comment]
Thanks for the report, however I believe this is working as intended. As far as I remember, `airbnb-base` uses the following config for `comma-dangle`, which means comma dangles will also be enforced for function invocations:
```js
    'comma-dangle': ['error', {
      arrays: 'always-multiline',
      objects: 'always-multiline',
      imports: 'always-multiline',
      exports: 'always-multiline',
      functions: 'always-multiline',
    }]
```

If you don't want to enforce trailing commas for functions invocations you should override the airbnb default config and turn off checking for functions:
```js
    'comma-dangle': ['error', {
      arrays: 'always-multiline',
      objects: 'always-multiline',
      imports: 'always-multiline',
      exports: 'always-multiline',
      functions: 'never',
    }]
```

**Comment by vitorbal** [comment]
ping @adispring, could you verify if my comment above solves your issue?

**Comment by adispring** [comment]
I think putting a comma after function's arguments is an syntax error

**Comment by ilyavolodin** [comment]
@adispring Trailing function commas is a stage 4 proposal, so it's not a syntax error if you have ES2017 enabled.

**Comment by adispring** [comment]
@vitorbal @ilyavolodin , got it, I'll close this issue.
